strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/pypi/PypiBrowsePackageHtmlResponseBuilderTest.java

```diff
@@ -0,0 +1,130 @@
+                + "        <head>\n"
+                + "            <title>Links for " + packageName + "</title>\n"
+                + "        </head>\n"
+                + "        <body>\n"
+                + "            <h1>Links for " + packageName + "</h1>\n"
+                + "                   " + links
+                + "        </body>\n"
+                + "    </html>";
+
+        List<Path> paths = new ArrayList<>();
+        paths.add(packagePath.normalize());
+        String htmlResponse = htmlResponseBuilder.getHtmlResponse(paths, packageName, repository);
+
+        assertThat(htmlResponse).isNotBlank().isEqualTo(expectedHtmlResponse);
+    }
+
+    @Test
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class, ArtifactManagementTestExecutionListener.class })
+    public void testMultiplePackagesFound(@PypiTestRepository(repositoryId = REPOSITORY_RELEASES, storageId = REPOSITORY_STORAGE) Repository repository,
+                                          @PypiTestArtifact(repositoryId = REPOSITORY_RELEASES, storageId = REPOSITORY_STORAGE, id = "hello_world", versions = { "1.0",
+                                                                                                                                                                 "2.0",
+                                                                                                                                                                 "3.0",
+                                                                                                                                                                 "4.0",
+                                                                                                                                                                 "5.0"
+                                          }) List<Path> packagePaths)
+        throws IOException
+    {
+
+        List<Path> paths = new ArrayList<>();
+        String links = "";
+        for (Path path : packagePaths)
+        {
+
+            paths.add(path.normalize());
+            PypiArtifactCoordinates artifactCoordinates = (PypiArtifactCoordinates) RepositoryFiles.readCoordinates((RepositoryPath) path.normalize());
+
+            links += "<a href=\"" + "/storages/" + repository.getStorage().getId() + "/" + repository.getId()
+                    + "/packages/" + artifactCoordinates.buildWheelPackageFileName() + "\">"
+                    + artifactCoordinates.buildWheelPackageFileName() + "</a><br>\n";
+        }
+
+        String packageName = "hello-world";
+        String expectedHtmlResponse = "<html>\n"
+                + "        <head>\n"
+                + "            <title>Links for " + packageName + "</title>\n"
+                + "        </head>\n"
+                + "        <body>\n"
+                + "            <h1>Links for " + packageName + "</h1>\n"
+                + "                   " + links
```

Comment:
There should probably be a check if the `packagePaths.isEmpty`, or not and only append these two lines, if there's actual contents...

Reply:
Is it possible that `packagePaths` is empty here?

Reply:
@carlspring  it seems we can resolve it now

Reply:
Thanks!